mixer/adapter/kubernetesenv/kubernetesenv_test.go

```diff
@@ -384,6 +392,83 @@ func TestKubegen_Generate(t *testing.T) {
 	}
 }
 
+func createMultiClusterSecret(k8s *fake.Clientset) error {
+	data := map[string][]byte{}
+	secret := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testSecretName,
+			Namespace: testSecretNameSpace,
+			Labels: map[string]string{
+				"istio/multiCluster": "true",
+			},
+		},
+		Data: map[string][]byte{},
+	}
+
+	data["testRemoteCluster"] = []byte("Test")
+	secret.Data = data
+	_, err := k8s.CoreV1().Secrets(testSecretNameSpace).Create(&secret)
+	return err
+}
+
+func deleteMultiClusterSecret(k8s *fake.Clientset) error {
+	var immediate int64
+
+	return k8s.CoreV1().Secrets(testSecretNameSpace).Delete(
+		testSecretName, &metav1.DeleteOptions{GracePeriodSeconds: &immediate})
+}
+
+func mockLoadKubeConfig(kubeconfig []byte) (*clientcmdapi.Config, error) {
+	return &clientcmdapi.Config{}, nil
+}
+
+func mockCreateInterfaceFromClusterConfig(clusterConfig *clientcmdapi.Config) (kubernetes.Interface, error) {
+	return fake.NewSimpleClientset(), nil
+}
+
+func Test_KubeSecretController(t *testing.T) {
+	secretcontroller.LoadKubeConfig = mockLoadKubeConfig
+	secretcontroller.CreateInterfaceFromClusterConfig = mockCreateInterfaceFromClusterConfig
```

Comment:
This seems to be redundant with line 433 in below.

Reply:
Needed to create a 'clientset' for secret creation (see line 410 above) plus a few other locations.